cmd/plugin/node_server.go

```diff
@@ -59,7 +61,9 @@ type NodeServer struct {
 }
 
 func (n NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (resp *csi.NodePublishVolumeResponse, err error) {
-	klog.Infof("mount request: %s", req.String())
+	u, _ := uuid.NewRandom()
+	valuesLogger := klog.LoggerWithValues(klog.NewKlogr(), "pod-name", req.VolumeContext["pod-name"], "namespace", req.VolumeContext["namespace"], "uid", req.VolumeContext["uid"], "request-id", u.String())
+	valuesLogger.Info(fmt.Sprintf("mount request: %s", req.String()))
```

Comment:
`valuesLogger.Info(fmt.Sprintf(...))` is messy but klogr doesn't support `infof()`.

Reply:
You might want to check `klog.SetLogFilter()` or [`klog.SetLogger()`](https://github.com/kubernetes/klog/blob/81f97ff8551e91cd32fbe508e2525db0c67da203/contextual.go#L42-L61) 

Another option is to use zap's [SugarLogger](https://github.com/uber-go/zap/blob/master/sugar.go) instead of klog's logger but I think that might need more discussion.

Reply:
I followed the advice [here](https://pkg.go.dev/github.com/go-logr/logr#readme-but-i-really-want-to-use-a-format-string) for converting format strings to structured logs. Open to discussion though